internal/command/views/operation.go

```diff
@@ -284,6 +290,19 @@ func (v *OperationJSON) Diagnostics(diags tfdiags.Diagnostics) {
 	v.view.Diagnostics(diags)
 }
 
+func resetSensitiveVariables(outputs map[string]jsonplan.Change, resourceChanges []jsonplan.ResourceChange) {
+	for i, output := range outputs {
+		output.BeforeSensitive = nil
+		output.AfterSensitive = nil
+		outputs[i] = output
+	}
+
+	for i := range resourceChanges {
+		resourceChanges[i].Change.BeforeSensitive = nil
+		resourceChanges[i].Change.AfterSensitive = nil
+	}
+}
```

Comment:
I'm not entirely sure the approach of altering the "Sensitive" values of outputs and resources in the input of the renderer is the best approach here. It might have some unwanted side-effects.

An example would be the following one. Assume you already have the following configuration:
```hcl
variable "var1" {
}

resource "null_resource" "a" {
  triggers = {
    a = var.var1
  }
}
```
And this configuration is applied with the value `"X"` for `var1`, and is in state. Later on, you change the configuration, and only change the variable to be sensitive:
```hcl
variable "var1" {
  sensitive = true
}

resource "null_resource" "a" {
  triggers = {
    a = var.var1
  }
}
```

Running a plan and providing the same value for `var1`, would result in an attempted change of the resource, including a warning that a field will now become sensitive:
```
OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

OpenTofu will perform the following actions:

  # null_resource.a will be updated in-place
  ~ resource "null_resource" "a" {
        id       = "3504371271674334511"
      ~ triggers = {
          # Warning: this attribute value will be marked as sensitive and will not
          # display in UI output after applying this change. The value is unchanged.
          ~ "a" = (sensitive value)
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

However, if you run `tofu plan --show-sensitive`, the plan does not mention that the field will become sensitive, and becomes unclear:
```
OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

OpenTofu will perform the following actions:

  # null_resource.a will be updated in-place
  ~ resource "null_resource" "a" {
        id       = "3504371271674334511"
        # (1 unchanged attribute hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

I'm not entirely sure that this is the only effect of overriding the sensitive values that are the input of the renderer. We should probably discuss what the expected behaviour here should be

Reply:
Well `RenderHuman` only affects the rendered output though. Currently, it can only tell the difference between a sensitive value and a non-sensitive value, and show a diff accordingly.

An arguably better approach would be to let it receive `showSensitive` as input, and then decide how to render the plan accordingly

Reply:
That's a good point @RLRabinowitz. It can be very confusing for the user to see an unexplainable change. Plus, afterward, when he runs commands without the `--show-sensitive` flag, he'll be surprised to see that the attribute is suddenly sensitive.
So IIMO this is unacceptable behaviour, and we should try and fix it.

Maybe instead of changing the `BeforeSensitive` and `AfterSensitive`, we can trickle down the `showSensitive` value as an input to the [sensitiveRenderer](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/renderers/sensitive.go#L25) (it's responsible for showing the `this attribute value will no longer be marked as sensitive` [warning](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/renderers/sensitive.go#L46)) and add a proper warning the fits our situation.
I don't think it should be hard to do that as the [sensitiveRenderer.RenderHuman()](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/renderers/sensitive.go#L32) already accepts [RenderHumanOpts](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/diff.go#L75) struct. From what I see we'll want to trickle down the `showSensitive` value similarly to how `Colorize` is passed (from the `jsonformat.Renderer` [here](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/views/operation.go#L105) to eventually the `RenderHumanOpts` [here](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/diff.go#L76)).

Reply:
I chose to implement resetSensitiveVariables in [/view/operation.go](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/views/operation.go#L293). My approach was to keep the showSensitive flag, which is used for view or display, separate from RenderHuman. I wanted this flag to be part of the view layer exclusively and to handle sensitive values within the view layer only.

Thank you for @RLRabinowitz pointing out that edge case. Resetting the values of `BeforeSensitive` and `AfterSensitive` also impacted [CheckForSensitive()](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/structured/sensitive.go#L50), where the [action](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/structured/sensitive.go#L87) was not updated due to the empty values of BeforeSensitive and AfterSensitive.

Reply:
Thanks, @Evi1Pumpkin , for your suggestion. Including the showSensitive flag in [jsonformat.Renderer](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/views/operation.go#L104) and using it in [RenderHumanOpts](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/jsonformat/computed/diff.go#L75) is a better approach. I tried this locally, and it worked as expected. I will update the PR accordingly once I've finalised the changes.